src/backend/vulkan/src/lib.rs

```diff
@@ -117,42 +122,152 @@ pub struct Instance {
 }
 
 fn map_queue_type(flags: vk::QueueFlags) -> QueueType {
-    if flags.subset(vk::QUEUE_GRAPHICS_BIT | vk::QUEUE_COMPUTE_BIT) { // TRANSFER_BIT optional
+    if flags.subset(vk::QueueFlags::GRAPHICS | vk::QueueFlags::COMPUTE) { // TRANSFER_BIT optional
         QueueType::General
-    } else if flags.subset(vk::QUEUE_GRAPHICS_BIT) { // TRANSFER_BIT optional
+    } else if flags.subset(vk::QueueFlags::GRAPHICS) { // TRANSFER_BIT optional
         QueueType::Graphics
-    } else if flags.subset(vk::QUEUE_COMPUTE_BIT) { // TRANSFER_BIT optional
+    } else if flags.subset(vk::QueueFlags::COMPUTE) { // TRANSFER_BIT optional
         QueueType::Compute
-    } else if flags.subset(vk::QUEUE_TRANSFER_BIT) {
+    } else if flags.subset(vk::QueueFlags::TRANSFER) {
         QueueType::Transfer
     } else {
         // TODO: present only queues?
         unimplemented!()
     }
 }
 
-extern "system" fn callback(
+extern "system" fn debug_utils_messenger_callback(
+    message_severity: vk::DebugUtilsMessageSeverityFlagsEXT,
+    message_type: vk::DebugUtilsMessageTypeFlagsEXT,
+    p_callback_data: *const vk::DebugUtilsMessengerCallbackDataEXT,
+    user_data: *mut libc::c_void,
```

Comment:
I don't think the extra `libc` dependency should be necessary for this – other functions already use `std::os::raw::c_void`